### PR TITLE
fix regex to more resiliently capture glob pattern

### PR DESCRIPTION
**Pull request by eleith**
when we wired up gerald to webapp, we were not able to get any rules to properly trigger reviews. we were using the following pattern in `REVIEWERS`

```
services/test-prep/** @Khan/test-prep
```

the syntax for the `REVIEWERS` file seems borrowed from github's `CODEOWNERS` file and so we had assumed that glob patterns and user names could be separated by any number of spaces.

however, the regex capture being used by gerald to parse the globs/names was requiring 3 or more spaces. 

this PR makes the following changes

- updates regex pattern capture to be more resilient to number of spaces/tabs
- adds a test to fail on main branch but passes with this PR
- format .babelrc
- updates package.json to install necessary dev packages to build
- adds a npm run build command to quickly compile/build in one step
- updates the npm run test to only compile the src dir
- updates gitignore for generated files

# TEST_PLAN

`npm run test` passes

### CODE REVIEW

**Comment by eleith** [comment]
fyi @jeresig / @dnerdy, this was tricky to debug, luckily actions can be pointed to branches and jobs can be re-run.

File: src/constants.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const MATCH_COMMENT_HEADER_REGEX = /^### (Reviewers:|Required reviewers:|
 
 // Gerald Files Regexes
 export const MATCH_REGEX_REGEX = /^"\/(.*?)\/([a-z]*)"$/;
-export const MATCH_PATTERN_REGEX = /(.(?!  @))*/;
+export const MATCH_PATTERN_REGEX = /(.(?! +@))*./;
```
Comment (jeresig) [inline src/constants.js:40]: OOF... thank you for catching this...

File: package.json [hunk 2]
```diff
@@ -1,10 +1,11 @@
 {
   "scripts": {
-    "dist": "ncc build src/pr-notify.js",
```
Comment (jeresig) [inline package.json:3]: huh... what was this doing before? It seems like what you have now is more accurate, at least.

**Review by jeresig — APPROVED** [review]
Thank you for fixing this!

**Pull request merged**